#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/Floor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/drinks/protos/Can.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/apartment_structure/protos/Wall.proto"

WorldInfo {
  basicTimeStep 16
}
Viewpoint {
  orientation -0.12933587133573451 -0.13295073907602117 0.982647614033111 4.722450109637431
  position 0.26235826187455935 1.6275323178550742 0.4024896196204394
  follow "robot:solid(2):solid"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Robot {
  translation 0.0009658014116646074 -3.873745271677173e-06 0.19667986994964823
  rotation 0.004633967536793347 0.9999889020483632 -0.0008497793696696017 0.003547128996846577
  children [
    Camera {
      translation 0.2 0 0.2
    }
    DEF Tyres Group {
      children [
        HingeJoint {
          jointParameters HingeJointParameters {
            position 118.95155844896719
            axis 0 1 0
            anchor 0 0.25 -0.15
          }
          device [
            RotationalMotor {
              name "wheel_left1"
            }
          ]
          endPoint Solid {
            translation -0.0005600160524789319 0.25002800048994067 -0.1499773656439623
            rotation 0.9281474706066957 -0.26287881948876846 0.2635090113650472 1.6483544319646832
            children [
              DEF wheel_shape Shape {
                appearance PBRAppearance {
                  baseColor 1 0.220691 0.0788586
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.05
                  radius 0.05
                }
              }
            ]
            boundingObject USE wheel_shape
            physics Physics {
            }
            linearVelocity 8.45184598163237e-09 -2.52880143416823e-09 5.909319288209573e-11
            angularVelocity -2.3821671843802234e-10 4.2896114729821873e-07 -1.2724199914182588e-08
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            position 63.04000550068769
            axis 0 1 0
            anchor -0.4 0.25 -0.15
          }
          device [
            RotationalMotor {
              name "wheel_left2"
            }
          ]
          endPoint Solid {
            translation -0.3999798580094451 0.2500639633922303 -0.1499270512453514
            rotation 0.9982742358199129 0.042521969918812644 -0.04050224897944155 1.5748076763429952
            children [
              DEF wheel_shape Shape {
                appearance PBRAppearance {
                  baseColor 1 0.220691 0.0788586
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.05
                  radius 0.05
                }
              }
            ]
            name "solid(3)"
            boundingObject USE wheel_shape
            physics Physics {
            }
            linearVelocity 8.441617121723729e-09 2.5566279419508997e-09 -5.870236632992459e-11
            angularVelocity -2.395531490243541e-10 -7.777558708902161e-09 -1.273136683879855e-08
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            position -2.2743354226509977e-06
            axis 0 1 0
            anchor 0 -0.25 -0.15
          }
          device [
            RotationalMotor {
              name "wheel_right1"
            }
          ]
          endPoint Solid {
            translation -1.025179308943844e-06 -0.2500600011721518 -0.14993824141204334
            rotation 0.996118963525891 -0.0613564836760054 0.06310619949751897 1.5771689066147085
            children [
              DEF wheel_shape Shape {
                appearance PBRAppearance {
                  baseColor 1 0.220691 0.0788586
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.05
                  radius 0.05
                }
              }
            ]
            name "solid(4)"
            boundingObject USE wheel_shape
            physics Physics {
            }
            linearVelocity 2.0745816073817922e-09 -2.5358879871341756e-09 -5.894250748587353e-11
            angularVelocity -2.3955518533289367e-10 5.910990413765834e-10 -1.2731229228505179e-08
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            position 6.775012204525501e-06
            axis 0 1 0
            anchor -0.39998 -0.250064 -0.149921
          }
          device [
            RotationalMotor {
              name "wheel_right2"
            }
          ]
          endPoint Solid {
            translation -0.39997997917465544 -0.25006400024400366 -0.14992703790764625
            rotation 0.9961195231433504 -0.061351773224273015 0.0631019455714436 1.5771683441936537
            children [
              DEF wheel_shape Shape {
                appearance PBRAppearance {
                  baseColor 1 0.220691 0.0788586
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.05
                  radius 0.05
                }
              }
            ]
            name "solid(5)"
            boundingObject USE wheel_shape
            physics Physics {
            }
            linearVelocity 2.0749598800081874e-09 2.5566467433879757e-09 5.858931613045453e-11
            angularVelocity -2.395354802717531e-10 -9.839477719866571e-11 -1.2731240729446014e-08
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 135.05579435647553
        axis 0 1 0
        anchor 0.22 0 0
      }
      device [
        RotationalMotor {
          name "sweeper_motor"
        }
      ]
      endPoint Solid {
        translation 0.22007826538915426 5.231219637471021e-05 -0.00019517259722416524
        rotation -8.859789192260602e-06 -0.9999999999600125 -1.2161947602867693e-06 3.180835184680952
        children [
          DEF d_left Solid {
            translation 0 0.15 0.15
            rotation 0 0 1 3.14
            children [
              DEF distance_sensor_shape Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  roughness 1
                  metalness 0
                }
                geometry Box {
                  size 0.03 0.02 0.02
                }
              }
              DistanceSensor {
                name "distance_sensor_left"
              }
            ]
          }
          DEF d_right Solid {
            translation -0.000509661 -0.17 0.15
            rotation 0 0 1 3.14
            children [
              DEF distance_sensor_shape Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  roughness 1
                  metalness 0
                }
                geometry Box {
                  size 0.03 0.02 0.02
                }
              }
              DistanceSensor {
                name "distance_sensor_right"
              }
            ]
          }
          DEF d_center Solid {
            translation -0.000175194 0.04 0.15
            rotation 0 0 1 3.14
            children [
              DEF distance_sensor_shape Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  roughness 1
                  metalness 0
                }
                geometry Box {
                  size 0.03 0.02 0.02
                }
              }
              DistanceSensor {
                name "distance_sensor_center"
                numberOfRays 2
              }
            ]
          }
          DEF d_center2 Solid {
            translation -0.000350391 -0.07 0.15
            rotation 0 0 1 3.14
            children [
              DEF distance_sensor_shape Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  roughness 1
                  metalness 0
                }
                geometry Box {
                  size 0.03 0.02 0.02
                }
              }
              DistanceSensor {
                name "distance_sensor_center_2"
                numberOfRays 2
              }
            ]
            name "solid(1)"
          }
          DEF sweeper_shape Shape {
            appearance PBRAppearance {
              roughness 1
              metalness 0
            }
            geometry Box {
              size 0.02 0.4 0.37
            }
          }
        ]
        name "solid(2)"
        boundingObject USE sweeper_shape
        physics Physics {
        }
        linearVelocity 5.29900755249636e-09 -5.308840355371211e-09 -1.879023926911712e-10
        angularVelocity -2.3947353165140143e-10 2.282494343345582e-08 -1.2730864024946946e-08
      }
    }
    Group {
      children [
        Solid {
          translation -1.85717e-06 8.4367e-17 -0.0301247
          rotation 2.8834614090237705e-08 -0.9999999999999996 -9.6654458917076e-10 9.827436276334678e-08
          children [
            DEF box_wall Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.02 0.4 0.3
              }
            }
          ]
          name "box_wall_1"
          boundingObject USE box_wall
        }
        Solid {
          translation -0.1 0.2 -6.2784e-06
          rotation 1.139551427790092e-15 -1.6955323793011871e-15 1 1.5699999999999998
          children [
            Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry DEF box_wall Box {
                size 0.02 0.6 0.4
              }
            }
          ]
          name "box_wall_2"
          boundingObject USE box_wall
        }
        Solid {
          translation -0.1 -0.2 -6.2784e-06
          rotation 1.139551427790092e-15 -1.6955323793011871e-15 1 1.5699999999999998
          children [
            Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry DEF box_wall Box {
                size 0.02 0.6 0.4
              }
            }
          ]
          name "box_wall_3"
          boundingObject USE box_wall
        }
        Solid {
          translation -0.4 -3.55826007184602e-21 -6.278396306458323e-06
          rotation 0.9807014561659081 -0.19272835386804643 -0.032869978542439886 2.889496319432257e-15
          children [
            DEF box_wall Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.02 0.4 0.4
              }
            }
          ]
          name "box_wall_4"
          boundingObject USE box_wall
        }
        Solid {
          translation -0.2 0 -0.18
          rotation 0 1 0 1.57
          children [
            DEF box_floor Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.02 0.4 0.4
              }
            }
          ]
          name "box_floor"
          boundingObject USE box_floor
          physics Physics {
            mass 20
          }
        }
      ]
    }
    SliderJoint {
      jointParameters JointParameters {
      }
      device [
        LinearMotor {
          name "slab_linear_motor"
        }
      ]
      endPoint Solid {
        rotation -3.765839153279543e-08 -0.9999999999999993 -2.498924693683275e-10 8.091120447688148e-07
        children [
          HingeJoint {
            jointParameters HingeJointParameters {
              position 0.00019391029432280074
              axis 0 1 0
              anchor 0 0 -0.19
            }
            device [
              RotationalMotor {
                name "slab_rotational_motor"
              }
            ]
            endPoint Solid {
              translation 0.12 6.6567e-16 -0.2
              rotation 0 1 0 0.00019391029416036622
              children [
                DEF sliding_slab_shape Shape {
                  appearance PBRAppearance {
                    roughness 1
                    metalness 0
                  }
                  geometry Box {
                    size 0.2 0.4 0.01
                  }
                }
              ]
              boundingObject USE sliding_slab_shape
            }
          }
        ]
        name "solid(1)"
      }
    }
  ]
  physics Physics {
  }
  controller "robot_controller"
  linearVelocity 5.25191076267404e-09 -1.1494422582714198e-10 -6.290985679481223e-11
  angularVelocity -2.395421891202899e-10 2.997210683667675e-10 -1.2731234131785772e-08
}
Floor {
}
Wall {
  translation 5 0 0
  size 0.2 10 1
}
Wall {
  translation -5 0 0
  name "wall(1)"
  size 0.2 10 1
}
Wall {
  translation 0 5 0
  rotation 0 0 1 1.5708
  name "wall(2)"
  size 0.2 10 1
}
Wall {
  translation 0 -5 0
  rotation 0 0 1 1.5708
  name "wall(3)"
  size 0.2 10 1
}
Can {
  translation 0.303242 -5.51504e-05 0.0321616
  rotation 0.5879148214864525 -0.5719958263200715 -0.5719938263206789 -2.078615307179586
}
